derive/src/rpc_attr.rs

```diff
@@ -113,6 +120,58 @@ impl RpcMethodAttribute {
 	}
 }
 
+fn validate_attribute_meta(meta: syn::Meta) -> Result<syn::Meta> {
+	#[derive(Default)]
+	struct Visitor {
+		meta_words: Vec<String>,
+		name_value_names: Vec<String>,
+		meta_list_names: Vec<String>,
+	}
+	impl<'a> Visit<'a> for Visitor {
+		fn visit_meta(&mut self, meta: &syn::Meta) {
+			match meta {
+				syn::Meta::List(list) => {
+					self.meta_list_names.push(list.ident.to_string())
+				},
+				syn::Meta::Word(ident) => self.meta_words.push(ident.to_string()),
+				syn::Meta::NameValue(nv) => self.name_value_names.push(nv.ident.to_string())
+			}
+		}
+	}
+
+	let mut visitor = Visitor::default();
+	visit::visit_meta(&mut visitor, &meta);
+
+	match meta.name().to_string().as_ref() {
+		RPC_ATTR_NAME => {
+			validate_idents(&meta, &visitor.meta_words, &[METADATA_META_WORD])?;
+			validate_idents(&meta, &visitor.name_value_names, &[RPC_NAME_KEY])?;
+			validate_idents(&meta, &visitor.meta_list_names, &[ALIASES_KEY])
+		},
+		PUB_SUB_ATTR_NAME => {
+			validate_idents(&meta, &visitor.meta_words, &[SUBSCRIBE_META_WORD, UNSUBSCRIBE_META_WORD])?;
+			validate_idents(&meta, &visitor.name_value_names, &[SUBSCRIPTION_NAME_KEY, RPC_NAME_KEY])?;
+			validate_idents(&meta, &visitor.meta_list_names, &[ALIASES_KEY])
+		},
+		_ => Ok(meta), // ignore other attributes - compiler will catch unknown ones
+	}
+}
+
+fn validate_idents(meta: &syn::Meta, attr_idents: &[String], valid: &[&str]) -> Result<syn::Meta> {
+	let invalid_meta_words: Vec<_> = attr_idents
+		.iter()
+		.cloned()
+		.filter(|w| !valid.iter().any(|v| v == w))
+		.collect();
```

Comment:
Isn't better to `filter()` before `clone()` to avoid less allocation?

However, I haven't checked to assembly but I suspect `rustc` is smart enough to re-arrange those